hudi-utilities/src/main/java/org/apache/hudi/utilities/config/KafkaSourceConfig.java

```diff
@@ -66,14 +66,6 @@ public class KafkaSourceConfig extends HoodieConfig {
       .markAdvanced()
       .withDocumentation("Schema to deserialize the records.");
 
-
-  public static final ConfigProperty<Long> KAFKA_FETCH_PARTITION_TIME_OUT = ConfigProperty
-      .key(PREFIX + "fetch_partition.time.out")
```

Comment:
Can you double check what is the assurance of this param when a time-out kicked in? Either from client or server side.

Reply:
This param was used in the only place - from client side in method `KafkaOffsetGen.fetchPartitionInfos(..)`. When this time-out kicked in, KafkaOffsetGen is not retrying to get `partitionsFor(topicName)` from KafkaConsumer anymore.  

But now we introduce HoodieRetyingKafkaConsumer, which already contains retrying mechanism in itself (with exponential backoff) and retrying policy is configured by MAX_RETRY_INTERVAL_MS, MAX_RETRY_INTERVAL_MS and MAX_RETRY_COUNT.

Reply:
> Can you double check what is the assurance of this param when a time-out kicked in? Either from client or server side.

or what do you mean by "assurance of this param"?

Reply:
I was thinking what the effect of that param to Kafka server, it looks like a client control from Hudi side, then we are good to migrate it.

Reply:
We should keep the legacy behavior too, that is the meaning of compatibility. Another choice is we log some warnings and tell user the new params have been taking effect.

Reply:
But these retrying behaviors are mutually exclusive. If the priority is to retain legacy behavior, then this whole PR doesn't make sense. So if you don't mind, I prefer the second choice and log some warnings soon.

Reply:
Either is fine for me.